pyccel/codegen/printing/ccode.py

```diff
@@ -814,8 +814,9 @@ def function_signature(self, expr, print_arg_names = True):
             arg_code = 'void'
         else:
             arg_code = ', '.join('{}'.format(self.function_signature(i, False))
-                        if isinstance(i, FunctionAddress) else '{0}{1}'.format(self.get_declare_type(i), i.name)
-                        if print_arg_names else '{}'.format(self.get_declare_type(i)) for i in args)
+                        if isinstance(i, FunctionAddress)
+                        else '{0}'.format(self.get_declare_type(i)) + i.name if print_arg_names else ''
```

Comment:
This line could use some parentheses

Reply:
Is it:
```
('{0}'.format(self.get_declare_type(i)) + i.name) if print_arg_names else ''
```
or
```
'{0}'.format(self.get_declare_type(i)) + (i.name if print_arg_names else '')
```
(This may be obvious but I don't know the operator precedence by heart